File: backend-v2/app/models/car_listing.ts
import { DateTime } from 'luxon'
import { BaseModel, column, belongsTo } from '@adonisjs/lucid/orm'
import User from '#models/user'
import type { BelongsTo } from '@adonisjs/lucid/types/relations'

export default class CarListing extends BaseModel {
  @column({ isPrimary: true })
  declare id: number

  @column()
  declare name: string

  @column()
  declare description: string

  @column()
  declare features: string[]

  @column()
  declare requirements: string[]

  @column()
  declare price: number

  @column()
  declare thumbnail: string

  @column()
  declare location: string

  @column()
  declare status: 'active' | 'rented' | 'hidden'

  @column()
  declare posterId: number

  @belongsTo(() => User, { foreignKey: 'posterId' })
  declare poster: BelongsTo<typeof User>

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime
}

File: app/wheels_up/lib/models/car_listing.dart
class CarListing {
  final int id;
  final String name;
  final String description;
  final double price;
  final String thumbnail;
  final List<String> features;
  final List<String> requirements;
  final String location;
  final DateTime createdAt;
  final DateTime updatedAt;
  final int posterId;

  CarListing({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.thumbnail,
    required this.features,
    required this.requirements,
    required this.location,
    required this.createdAt,
    required this.updatedAt,
    required this.posterId,
  });

  factory CarListing.fromJson(Map<String, dynamic> json) {

    return CarListing(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      thumbnail: json['thumbnail'],
      features: List<String>.from(json['features']),
      requirements: List<String>.from(json['requirements']),
      location: json['location'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      posterId: json['posterId'],
    );
  }
}

class CarListing2 {
  final String id;
  final String name;
  final String description;
  final double pricePerHour;
  final String thumbnail;
  final List<String> requirements;
  final String location;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String posterId;

  CarListing2({
    required this.id,
    required this.name,
    required this.description,
    required this.pricePerHour,
    required this.thumbnail,
    required this.requirements,
    required this.location,
    required this.createdAt,
    required this.updatedAt,
    required this.posterId,
  });

  factory CarListing2.fromJson(Map<String, dynamic> json) {

    return CarListing2(
      id: json['id'],
      name: json['title'],
      description: json['description'],
      pricePerHour: json['pricePerHour'].toDouble(),
      thumbnail: json['thumbnail'],
      requirements: List<String>.from(json['requirements']),
      location: json['location'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      posterId: json['posterId'],
    );
  }
}
